### PR TITLE
Add test for VMware PXE discovery provisioning

### PR DESCRIPTION
**Pull request by shubhamsg199**
Component Audit: Adding test for VMware PXE discovery provisioning

### CODE REVIEW

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8104
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery --external-logging --include-onprem-provisioning
Test Result : =========== 8 deselected, 31 warnings, 2 errors in 771.16s (0:12:51) ===========
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8393
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery --external-logging --include-onprem-provisioning
Test Result : ========== 8 deselected, 45 warnings, 2 errors in 1119.83s (0:18:39) ===========
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8394
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery --external-logging --include-onprem-provisioning
Test Result : ========== 8 deselected, 38 warnings, 2 errors in 1110.27s (0:18:30) ===========
```

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8407
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery --external-logging --include-onprem-provisioning
Test Result : ========== 2 passed, 8 deselected, 114 warnings in 2674.05s (0:44:34) ==========
```

File: pytest_fixtures/component/provision_vmware.py [hunk 1]
```diff
@@ -107,3 +109,19 @@ def module_vmware_image(
...
+    vm_boot_firmware = 'efi' if pxe_loader.vm_firmware == 'uefi' else 'bios'
+    host_data = Broker(
+        workflow='deploy-blank-vm-vcenter',
+        artifacts='last',
+        vm_network=settings.provisioning.vlan_id,
+        vm_boot_firmware=vm_boot_firmware,
+    ).execute()
+    yield host_data
+    # delete the host
+    vmware_host = VMWareVirtualMachine(vmwareclient, name=f'{host_data["name"]}')
```
Comment (Gauravtalreja1) [inline pytest_fixtures/component/provision_vmware.py:126]:
```suggestion
    vmware_host = VMWareVirtualMachine(vmwareclient, name=host_data['name'])
```

File: pytest_fixtures/component/provision_vmware.py [hunk 2]
```diff
@@ -107,3 +109,19 @@ def module_vmware_image(
...
+    vmware_host.delete()
```
Comment (Gauravtalreja1) [inline pytest_fixtures/component/provision_vmware.py:127]: how about confirming after delete if host is really deleted on vcenter with wrapanapi? like 
`assert not VMWareVirtualMachine(vmwareclient, name=host_data['name'])`

File: tests/foreman/api/test_computeresource_vmware.py [hunk 3]
```diff
@@ -124,3 +125,60 @@ def test_positive_provision_end_to_end(
         delay=10,
     )
     assert host.read().build_status_label == 'Installed'
+
+
+@pytest.mark.e2e
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:130]: we already cover normal discovery in e2e, so I think we can exclude this test for e2e

File: tests/foreman/api/test_computeresource_vmware.py [hunk 4]
```diff
@@ -124,3 +125,60 @@ def test_positive_provision_end_to_end(
...
+        timeout=1500,
+        delay=40,
+    )
+    discovered_host = sat.api.DiscoveredHost().search(query={'mac': mac})[0]
+    discovered_host.hostgroup = provisioning_hostgroup
+    discovered_host.location = provisioning_hostgroup.location[0]
+    discovered_host.organization = provisioning_hostgroup.organization[0]
+    discovered_host.build = True
+    host = discovered_host.update(['hostgroup', 'build', 'location', 'organization'])
+    host = sat.api.Host().search(query={"search": f'name={host.name}'})[0]
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:176]:
```suggestion
    host = sat.api.Host().search(query={'search': f'name={host.name}'})[0]
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 5]
```diff
@@ -124,3 +125,60 @@ def test_positive_provision_end_to_end(
...
+
+    :steps:
+        1. Boot up the host to discover
+        2. Provision the host
+
+    :expectedresults: Host should be provisioned successfully
+
+    """
+    # start the provisioning host
+    vmware_host = VMWareVirtualMachine(vmwareclient, name=f'{provisioning_vmware_host["name"]}')
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:161]:
```suggestion
    vmware_host = VMWareVirtualMachine(vmwareclient, name=provisioning_vmware_host['name'])
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 6]
```diff
@@ -124,3 +125,60 @@ def test_positive_provision_end_to_end(
...
+    vmware_host.start()
+    mac = provisioning_vmware_host['provisioning_nic_mac_addr']
+    sat = module_discovery_sat.sat
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:164]: Can you move this contants at start of the test?

File: tests/foreman/api/test_computeresource_vmware.py [hunk 7]
```diff
@@ -124,3 +125,60 @@ def test_positive_provision_end_to_end(
         delay=10,
     )
     assert host.read().build_status_label == 'Installed'
+
+
+@pytest.mark.e2e
+@pytest.mark.on_premises_provisioning
+@pytest.mark.parametrize('module_provisioning_sat', ['discovery'], indirect=True)
+@pytest.mark.parametrize('pxe_loader', ['bios', 'uefi'], indirect=True)
+@pytest.mark.rhel_ver_list([8])
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:134]:
```suggestion
@pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
```

File: pytest_fixtures/component/provision_vmware.py [hunk 8]
```diff
@@ -107,3 +109,19 @@ def module_vmware_image(
         uuid=settings.vmware.image_name,
         password=settings.provisioning.host_root_password,
     ).create()
+
+
+@pytest.fixture
+def provisioning_vmware_host(pxe_loader, vmwareclient):
+    """Fixture to check out blank VM on VMware"""
+    vm_boot_firmware = 'efi' if pxe_loader.vm_firmware == 'uefi' else 'bios'
+    host_data = Broker(
```
Comment (Gauravtalreja1) [inline pytest_fixtures/component/provision_vmware.py:118]:
```suggestion
    provisioning_host = Broker(
```

Comment (shubhamsg199) [reply pytest_fixtures/component/provision_vmware.py]: updated

File: tests/foreman/api/test_computeresource_vmware.py (hunk 7 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_computeresource_vmware.py]: updated

File: tests/foreman/api/test_computeresource_vmware.py (hunk 3 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_computeresource_vmware.py]: updated

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8492
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery --external-logging --include-onprem-provisioning
Test Result : =========== 8 deselected, 15 warnings, 2 errors in 187.55s (0:03:07) ===========
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8494
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery --external-logging --include-onprem-provisioning
Test Result : =========== 8 deselected, 15 warnings, 2 errors in 141.02s (0:02:21) ===========
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8495
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery --external-logging --include-onprem-provisioning
Test Result : =========== 8 deselected, 15 warnings, 2 errors in 135.58s (0:02:15) ===========
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8496
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_positive_provision_vmware_pxe_discovery --external-logging --include-onprem-provisioning
Test Result : ========== 2 passed, 8 deselected, 116 warnings in 2733.22s (0:45:33) ==========
```

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**